### DISCUSSION TITLE
EDDI does not speak in configured language

### DISCUSSION THREAD

**Discussion by tbayart** [EDCD/EDDI#2406] [EDDI Standalone Q&A] [answered]
Hi here,

I'm french on a french Windows 10 but i'm running all my programs in english, including Elite Dangerous.
EDDI is setup for english and all texts are displayed in english.
In Speech Responder, i checked a few scripts and most of them are in english.

I enabled speechresponder.out
it contains french phrases but the speech is english so it's really really weird

I want to get english phrases for proper english voice
What am i doing wrong ?

**Comment by Tkael** [comment]
> Hi here,
> 
> I'm french on a french Windows 10 but i'm running all my programs in english, including Elite Dangerous. EDDI is setup for english and all texts are displayed in english. In Speech Responder, i checked a few scripts and most of them are in english.
> 
> I enabled speechresponder.out it contains french phrases but the speech is english so it's really really weird
> 
> I want to get english phrases for proper english voice What am i doing wrong ?

Hi there, 
1. Are you using the default personality? If not, does the issue occur with the default personality?
2. Have you identified the scripts which are being spoken which contain french phrases? Are they customized scripts or default scripts? 
3. Could you please include screen shots of your language setting and Text-to-Speech settings?
4. Are you using VoiceAttack?

**Comment by tbayart** [comment]
1. I made a copy of default personality without changing anything except disabling some events. I changed back to default and restarted EDDI the issue is still happening.
2. All scripts speaks french, i only have default ones.
3. I attached the screenshots
![EDDI settings](https://user-images.githubusercontent.com/5262768/195882808-bb95ff2a-cd12-420e-8b7d-31f5a2e62839.png)
![EDDI TTS settings](https://user-images.githubusercontent.com/5262768/195882813-5e7d4df7-8596-421d-bfbb-d316aa10bfeb.png)
4. I first started using EDDI with VoiceAttack and tried also as a standalone. I thought i faced the issue in both modes but i tried again and in standalone mode EDDI is speaking in english. So the issue is when using EDDI with VoiceAttack.
I'm sorry, i should have posted this under **EDDI VoiceAttack Q&A**

**Reply by Tkael** [reply]
No worries, thank you for the information. 
It seems like a language setting is being carried over from VoiceAttack.

Please check the value of the following tokens in VoiceAttack?

{STATE_CULTURE}
{STATE_UICULTURE}
{STATE_SPEECHCULTURE}

I'm not at my desk but I think there may be a setting in VoiceAttack to allow you to change these values?

**Reply by tbayart** [reply]
All three tokens are resolved to fr-FR
I checked Windows speech engine settings and its english US
![Windows TTS settings](https://user-images.githubusercontent.com/5262768/195922206-3d035148-edb5-4988-9e89-6c8b082a1757.png)

Maybe i should ask on VoiceAttack forum ?

**Reply by Tkael** [reply]
It has to do with language configuration inheritance between application threads, but we can probably handle it without involving Gary by manually updating the language configuration on the affected threads.
Is the language leak happening on events, on commands that you invoke from VoiceAttack, or on both?
Have you noticed any French variable values from EDDI variables in VoiceAttack?

**Reply by tbayart** [reply]
On events for sure.

I suppose this is related to the script used for generating speech text. 
So if the script set some localized value into a variable, the value will probably be in French.
I'm very new to EDDI so maybe i misunderstood the way speech text is generated.

I have to test this, first find a command with some localized text, then implement it to see what happen.
Feel free to tell me if you have any suggestion.

**Reply by Tkael** [reply]
If you are using the EDDI profile in VoiceAttack, you are able to write a large amount of EDDI information that you can check for French outputs:
![image](https://user-images.githubusercontent.com/4382539/196054490-d689045f-0ca4-4138-ad3d-a64703eadc19.png)

**Comment by tbayart** [comment]
All EDDI information commands give english output
![screen capture](https://user-images.githubusercontent.com/5262768/196182358-2b413d9b-f2dc-44db-a570-e2e5bb8eab80.png)

**Comment by Tkael** [comment, marked as answer]
I'm moving this to an issue, #2407.
